File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.c
long long getDescentPeriods (int* prices, int pricesSize) {
    long long sum = 1, des = 1;
    for (int i = 1; i < pricesSize; i++) {
        des = (prices [i] + 1 == prices [i - 1]) * des + 1;
        sum += des;
    }
    return sum;
}

File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.py
class Solution:
    def getDescentPeriods (self, prices : List [int]) -> int:
        summ, des, prev = 0, 0, -1
        for x in prices:
            des = (-((x + 1) == prev) & des) + 1
            summ += des
            prev = x
        return summ